pkg/jobrunaggregator/jobruntestcaseanalyzer/analyzer.go

```diff
@@ -0,0 +1,278 @@
+package jobruntestcaseanalyzer
+
+import (
+	"context"
+	"encoding/xml"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"cloud.google.com/go/storage"
+
+	"k8s.io/apimachinery/pkg/util/clock"
+
+	"github.com/openshift/ci-tools/pkg/jobrunaggregator/jobrunaggregatorapi"
+	"github.com/openshift/ci-tools/pkg/jobrunaggregator/jobrunaggregatorlib"
+	"github.com/openshift/ci-tools/pkg/junit"
+)
+
+const (
+	installTest = "cluster install should succeed"
+)
+
+type requiredPasses struct {
+	testName               string
+	requiredNumberOfPasses int
+}
+
+// JobRunTestCaseAnalyzerOptions
+// 1. get a list of jobs from big query that meet the passed criteria: platform, network etc.
+// 2. finds jobruns for matching jobs for the specified payload tag
+// 3. reads all junit for the each jobrun
+// 4. constructs a synthentic junit that check minimum required succuessful passes for configured tests
+type JobRunTestCaseAnalyzerOptions struct {
+	payloadTag          string
+	workingDir          string
+	jobRunStartEstimate time.Time
+	clock               clock.Clock
+	timeout             time.Duration
+	platform            string
+	infrastructure      string
+	network             string
```

Comment:
these are used only for job name selection?  if so, please create a 

```go
type JobNameSelector interface{
    ListJobNames(ctx)
```

Reply:
Not really. They are also used to create specific test names later. So it makes sense to have them as variables.

Reply:
> Not really. They are also used to create specific test names later. So it makes sense to have them as variables.

Then perhaps doc-ing them is appropriate